### ISSUE TITLE
[BUG] backup restore is not done with the right URL

### ISSUE THREAD

**Issue by n0rad** [longhorn/longhorn#3336]
Using 1.2.2 version
To backup safely in multiple places, I rotate the `backup-target` to different NFS every 10 days.

Longhorn is caching backups and do store poll interval.  When the target is changes, Old target backups are seen in UI in error, new target backups are added.
When restoring a specific backup version targeting the new NFS, while the UI still have old target, **longhorn is trying to restore the backup with the old NFS target**
 
**my case :**

- My backup target is change from `nfs-backup-srv3.backup` to `nfs-backup-srv1.backup`.
- triggering a backup for volume XXXX arrives on `nfs-backup-srv1.backup`
- backup UI for volume XXXX list some versions in error (from the old target) and the new backup with the URL : `nfs://longhorn-nfs-backup-srv1.backup:/?backup=backup-bd302e325e6c4e99&volume=XXXX `
- I trigger a restore using the restore button on this version

logs is telling :
```
longhorn-manager-25h5g longhorn-manager E1129 00:25:21.254934       1 engine_controller.go:695] failed to update status for engine XXXX-e-b8b066ea: failed to restore backup backup-bd302e325e6c4e99 in engine monitor, will retry the restore later: failed to execute: /var/lib/longhorn/engine-binaries/longhornio-longhorn-engine-v1.2.2/longhorn [--url 10.42.1.243:10013 backup restore nfs://longhorn-nfs-backup-srv3.backup:/?backup=backup-bd302e325e6c4e99&volume=XXXX], output failed to get the current restoring backup info: Cannot mount nfs longhorn-nfs-backup-srv3.backup:/: Timeout executing: mount [-t nfs4 -o nfsvers=4.2 -o actimeo=1 longhorn-nfs-backup-srv3.backup:/ /var/lib/longhorn-backupstore-mounts/longhorn-nfs-backup-srv3_backup], output , error <nil>
```

it's trying to restore the good backup version, but with the wrong target, not using the full url given by the `get URL` button

**Comment by derekbit** [comment]
@n0rad 
How do you rotate the backup-target?

> backup UI for volume XXXX list some versions in error (from the old target)

Which field show the error?

**Comment by jenting** [comment]
- How do you perform switching the backup target? By manually editing it on Longhorn UI _or_ by a script to change the Setting CR?
- What is the old target backups in error means? Is it means the Backup with `Error` state?
- The `get URL` button and the `restore` button should use the same URL from the Backup CR, we need to double-check the UI code.

**Comment by n0rad** [comment]
- I use a cronjob that rotate the lhs setting value. 
I think it was working more smoothly in previous versions when I put in place the rotation (1.0.X if I remember well)

- in the UI there is a line for versions that are fully in error (creation date, state, size, ... : 
 ```
backup-12e6e28a693b41e5 | XXXX | Error |   | 0 Bi |   |   | Invalid date |  
```
I think it's a separated, but linked bug due to the fact that I changed the target url. What is in cache cannot be updated because the endpoint is not mounted (and will not be mounted since it's not the target anymore)
ex :
```
backup=backup-911094e759984219 controller=longhorn-backup error="error getting backup config nfs://longhorn-nfs-backup-srv3.backup:/?backup=backup-911094e759984219&volume=XXXX: Timeout executing: /var/lib/longhorn/engine-binaries/longhornio-longhorn-engine-v1.2.2/longhorn [backup inspect nfs://longhorn-nfs-backup-srv3.backup:/?backup=backup-911094e759984219&volume=XXXX], output , stderr, , error <nil>" node=srv1
```
I think the restore is wrong due to those versions staying in error, even if the restore should not use this target.

- this is why I created the issue, the UI is really telling me that the backup is on `srv1`, while the restore error log is trying to get the version from `srv3`, and the restore progress bar never start

**Comment by jenting** [comment]
Yes, that might be a case that during backup creation in progress. If the backup target changes, the backup falls into the error state because the NFS mount point is gone.

I'd like to know about the restore error, could you please dump the Backup CR `backup-911094e759984219` to us?
```shell
kubectl get lhb backup-911094e759984219 -n longhorn-system -o yaml
```

**Comment by n0rad** [comment]
I made sure I changed the target before triggering the backup & triggering the restore. 
Even if it was the case, this version backup only exists on the new target. it should not restore it from the old target :thinking: 

for the backuo version in error : 
```yaml
apiVersion: longhorn.io/v1beta1
kind: Backup
metadata:
  creationTimestamp: "2021-11-26T03:00:10Z"
  finalizers:
  - longhorn.io
  generation: 1
  labels:
    backup-volume: XXXX
  name: backup-911094e759984219
  namespace: longhorn-system
  resourceVersion: "63803930"
  uid: bcc13051-e58c-461d-873e-2c0d7e654002
spec:
  labels:
    KubernetesStatus: '{"pvName":"syncthing-media-1","pvStatus":"Bound","namespace":"media","pvcName":"XXXX","lastPVCRefAt":"","workloadsStatus":[{"podName":"XXXX","podStatus":"Running","workloadName":"XXXX","workloadType":"StatefulSet"}],"lastPodRefAt":""}'
    RecurringJob: c-85d3nh
  snapshotName: c-85d3nh-c-0f1c2702
  syncRequestedAt: "0001-01-01T00:00:00Z"
status:
  backupCreatedAt: ""
  labels: null
  lastSyncedAt: "0001-01-01T00:00:00Z"
  messages: null
  ownerID: srv1
  size: ""
  snapshotCreatedAt: ""
  snapshotName: ""
  state: Error
  url: ""
  volumeBackingImageName: ""
  volumeCreated: ""
  volumeName: ""
  volumeSize: ""

```
